runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -4756,6 +4756,17 @@ static TR::Register * generateMultianewArrayWithInlineAllocators(TR::Node *node,
    bool use64BitClasses = comp->target().is64Bit() && !TR::Compiler->om.generateCompressedObjectHeaders();
 
    generateRXInstruction(cg, use64BitClasses ? TR::InstOpCode::STG : TR::InstOpCode::ST, node, classReg, generateS390MemoryReference(targetReg, TR::Compiler->om.offsetOfObjectVftField(), cg));
+
+   static char * disableBatchClear = feGetEnv("TR_DisableBatchClear");
+   // If batch clear is disabled, set element's size and mustBeZero ('0') fields to 0
+   if (disableBatchClear)
+      {
+      cursor = generateRXInstruction(cg, TR::InstOpCode::ST, node, firstDimLenReg, generateS390MemoryReference(targetReg, fej9->getOffsetOfContiguousArraySizeField(), cg));
+      iComment("Init 1st dim size field.");
+      cursor = generateRXInstruction(cg, TR::InstOpCode::ST, node, firstDimLenReg, generateS390MemoryReference(targetReg, fej9->getOffsetOfDiscontiguousArraySizeField(), cg));
```

Comment:
seemingly the ContiguousArraySize and DiscontiguousArraySize fields are adjacent, I can ad a check here and if it was true, use a 64bit command on 64bit system to clear both fields with one instruction but not sure if it worth it.

Reply:
I used `XC` instruction first but since we already have a zero register (firstDimLenReg) here, decided to use `ST`.

Reply:
Understandingly the setting of these two fields to zero is inspired from x86 implementation- But I would like to know reason. Especially when I see the offset of size for contiguous array size and discontinuous array size, we write 4 bytes, while the offset of mustBeZero field is would be one after size.

Reply:
Based on the implementation of [ArratLengthEvaluator](https://github.com/eclipse-openj9/openj9/blob/671e76243d8e011486e26f4d2d298012efe2e2d9/runtime/compiler/z/codegen/J9TreeEvaluator.cpp#L5040), we check the contiguous field to get the size of an array, if it is zero, then we check the discontiguous field. So in case of zero size array, both must be zero.

Reply:
[This chart](https://github.com/eclipse-openj9/openj9/blob/671e76243d8e011486e26f4d2d298012efe2e2d9/runtime/compiler/x/codegen/J9TreeEvaluator.cpp#L7008C7-L7008C45) make it more clear why we have two size fields and why discontiguous size part is name 'mustBeZero' !